includes/admin/payments/actions.php

```diff
@@ -262,24 +262,34 @@ function edd_trigger_purchase_delete( $data ) {
 add_action( 'edd_delete_payment', 'edd_trigger_purchase_delete' );
 
 /**
- * New in 3.0, destroys an order, and all it's data, and related data.
+ * New in 3.0, permanently destroys an order, and all its data, and related data.
  *
  * @since 3.0
  *
- * @param $data
+ * @param array $data Arguments passed.
  */
 function edd_trigger_destroy_order( $data ) {
+
 	if ( wp_verify_nonce( $data['_wpnonce'], 'edd_payment_nonce' ) ) {
 
 		$payment_id = absint( $data['purchase_id'] );
 
 		if ( ! current_user_can( 'delete_shop_payments', $payment_id ) ) {
-			wp_die( __( 'You do not have permission to edit this payment record', 'easy-digital-downloads' ), __( 'Error', 'easy-digital-downloads' ), array( 'response' => 403 ) );
+			wp_die( esc_html_e( 'You do not have permission to edit this payment record', 'easy-digital-downloads' ), esc_html_e( 'Error', 'easy-digital-downloads' ), array( 'response' => 403 ) );
 		}
 
 		edd_destroy_order( $payment_id );
 
-		edd_redirect( admin_url( '/edit.php?post_type=download&page=edd-payment-history&edd-message=payment_deleted' ) );
+		$redirect_link = add_query_arg(
+			array(
+				'post_type'   => 'download',
+				'page'        => 'edd-payment-history',
+				'edd-message' => 'payment_deleted',
+				'status'      => 'trash',
+			),
+			admin_url( 'edit.php' ),
```

Comment:
Although this is totally working as it is, we could use `edd_get_admin_base_url` here instead of the `admin_url` function and remove the `post_type` parameter.

Reply:
I thought about that, but isn't `edd_get_admin_base_url` a 3.0 function?  I wasn't sure about having to add an if/else here.

Reply:
Yes, but it's fine since this is going directly into 3.0 itself instead of an extension--it's the one time we can be sure the function will exist. No condition needed. :)